requestmatch/matcher_test.go

```diff
@@ -1093,6 +1090,44 @@ func TestFreeWildcardParamWithSlash(t *testing.T) {
 	}
 }
 
+func TestHeaderMatchCaseInsensitiveTo(t *testing.T) {
+	routes, err := eskip.Parse(`Header("some-header", "some-value") -> "https://example.org"`)
+	if err != nil {
+		t.Error(err)
+	}
+
+	rd := &def{routes[0]}
+
+	m, errs := Make([]Definition{rd}, false)
+	if len(errs) != 0 {
+		t.Error("failed to make matcher")
+	}
+
+	r, _ := m.Match(&http.Request{URL: &url.URL{}, Header: http.Header{"Some-Header": []string{"some-value"}}})
+	if r == nil {
+		t.Error("failed to match header, case insensitive, to")
+	}
+}
+
+func TestHeaderMatchCaseInsensitiveFrom(t *testing.T) {
+	routes, err := eskip.Parse(`Header("Some-Header", "some-value") -> "https://example.org"`)
+	if err != nil {
+		t.Error(err)
+	}
+
+	rd := &def{routes[0]}
+
+	m, errs := Make([]Definition{rd}, false)
+	if len(errs) != 0 {
+		t.Error("failed to make matcher")
+	}
+
+	r, _ := m.Match(&http.Request{URL: &url.URL{}, Header: http.Header{"some-header": []string{"some-value"}}})
```

Comment:
considered this in part in the cleanup branch. There is this reoccurring topic of DRY in unit tests, too dry may hide the little story being tested. In some cases in that branch, I cleaned the tests up to be a bit more DRY, as it is getting more stable slowly, but actually this matcher_test stayed quite similar as what we see here. am I getting it very wrong? should I keep the tests less copy-pasty?